conda/lock.py

```diff
@@ -61,11 +73,44 @@ def __enter__(self):
                 return self
 
         stdoutlog.error("Exceeded max retries, giving up")
-        raise LockError(lockstr % self.lock_path)
+        raise LockError(LOCKSTR % self.lock_path)
 
     def __exit__(self, exc_type, exc_value, traceback):
         try:
             os.rmdir(self.lock_path)
             os.rmdir(self.path)
         except OSError:
             pass
+
+
+class FileLock(object):
+    """
+    Context manager to handle locks.
+    """
+    def __init__(self, filepath, retries=10):
+        self.filepath = abspath(filepath)
+        self.retries = retries
+
+    def __enter__(self):
+        sleeptime = 1
+        self.lockpath = "{0}.pid{1}.{2}".format(self.filepath, os.getpid(), LOCKFN)
+        lockglobstr = "{0}.pid*.{2}".format(self.filepath, LOCKFN)
+        lastglobmatch = None
+        for _ in range(self.retries):
+            globresult = glob(lockglobstr)
+            if globresult:
+                stdoutlog.info(LOCKSTR % globresult[0])
```

Comment:
I'm assuming here that glob is only ever going to have a single element, which is incredibly lazy, but for this demo code I didn't want to change the structure of `LOCKSTR`.